app/src/org/commcare/xml/CommCareTransactionParserFactory.java

```diff
@@ -21,9 +21,22 @@
 import android.content.Context;
 
 /**
- * @author ctsims
+ * The CommCare Transaction Parser Factory wraps all of the current
+ * transactions that CommCare knows about, and provides the appropriate hooks
+ * for parsing through XML and dispatching the right handler for each
+ * transaction.
+ *
+ * It should be the central point of processing for transactions (eliminating
+ * the use of the old datamodel based processors) and should be used in any
+ * situation where a transaction is expected to be present.
  *
+ * It is expected to behave more or less as a black box, in that it directly
+ * creates/modifies the data models on the system, rather than producing them
+ * for another layer or processing.
+ *
+ * @author ctsims

```

Comment:
Copied this comment over from commcare repo's application/src/org/commcare/util/CommCareTransactionParserFactory.java.